src/compiler/scanner.ts

```diff
@@ -2117,8 +2122,9 @@ namespace ts {
             }
 
             if (isIdentifierStart(ch, ScriptTarget.Latest)) {
-                while (isIdentifierPart(text.charCodeAt(pos), ScriptTarget.Latest) && pos < end) {
-                    pos++;
+                let nextChar: number;
+                while (isIdentifierPart(nextChar = codePointAt(text, pos), ScriptTarget.Latest) && pos < end) {
+                    pos += charSize(nextChar);

```

Comment:
stupid question: why isn't `charSize(c) === 2 when c >= 0x10000` instead of `c > 0x10000`?